python/lsst/pipe/tasks/computeExposureSummaryStats.py

```diff
@@ -471,6 +502,97 @@ def update_masked_image_stats(self, summary, masked_image):
         meanVar, _ = statObj.getResult(afwMath.MEANCLIP)
         summary.meanVar = meanVar
 
+    def update_effective_time_stats(self, summary, exposure):
+        """Compute effective exposure time statistics to estimate depth.
+
+        The effective exposure time is the equivalent shutter open
+        time that would be needed under nominal conditions to give the
+        same signal-to-noise for a point source as what is achieved by
+        the observation of interest. This metric combines measurements
+        of the point-spread function, the sky brightness, and the
+        transparency.
+
+        .. _teff_definitions:
+
+        The effective exposure time and its subcomponents are defined in [1]_
+
+        References
+        ----------
+
+        .. [1] Neilsen, E.H., Bernstein, G., Gruendl, R., and Kent, S. (2016).
+               Limiting Magnitude, \tau, teff, and Image Quality in DES Year 1
+               https://www.osti.gov/biblio/1250877/
+
+
+        Parameters
+        ----------
+        summary : `lsst.afw.image.ExposureSummaryStats`
+            Summary object to update in-place.
+        exposure : `lsst.afw.image.ExposureF`
+            Exposure to grab band and exposure time metadata
+
+        """
+        self.log.info("Updating effective exposure time")
+
+        nan = float("nan")
+        summary.effTime = nan
+        summary.effTimeScalePsfSigma = nan
+        summary.effTimeScaleSkyBg = nan
+        summary.effTimeScaleZeroPoint = nan
+
+        exposureTime = exposure.getInfo().getVisitInfo().getExposureTime()
+        filterLabel = exposure.getFilter()
+        if (filterLabel is None) or (not filterLabel.hasBandLabel):
+            band = None
+        else:
+            band = filterLabel.bandLabel
+
+        # PSF component
+        if np.isnan(summary.psfSigma):
+            self.log.warn("PSF sigma is NaN")
```

Comment:
I don't love having a warning here.  Presumably we'll have warnings upstream for bad psfs already, and doubling them up is not usually helpful.  I'd even make this a `debug`.